### PR TITLE
Bluetooth: controller: Add LE Read Buffer Size v2 support

### PR DESCRIPTION
**Pull request by cvinayak**
Add LE Read Buffer Size v2 support in the controller
implementation.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Review by thoh-ot — APPROVED** [review]

**Review by jhedberg — CHANGES_REQUESTED** [review]
Looks fine, but could you split the `hci.h` renames which cause updates to the host side into a separate commit? (it's not really proper to have host changes in a commit starting with "Bluetooth: controller: ..")

File: subsys/bluetooth/controller/hci/hci.c [hunk 1]
```diff
@@ -1156,6 +1162,32 @@ static void le_read_buffer_size(struct net_buf *buf, struct net_buf **evt)
...
+#if defined(CONFIG_BT_CTLR_ISO_TX_BUFFERS)
+#define BT_CTLR_ISO_TX_BUFFERS CONFIG_BT_CTLR_ISO_TX_BUFFERS
+#else
+#define BT_CTLR_ISO_TX_BUFFERS 0
+#endif
+#if defined(CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE)
+#define BT_CTLR_ISO_TX_BUFFER_SIZE CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE
+#else
+#define BT_CTLR_ISO_TX_BUFFER_SIZE 0
+#endif
```
Comment (Thalley) [inline subsys/bluetooth/controller/hci/hci.c:1175]: Why is this necessary? `BT_CTLR_ISO_TX_BUFFER_SIZE` is not conditional on anything, so it's always defined.

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/hci/hci.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/hci/hci.c:1175]: oh, `BT_CTLR_ISO_TX_BUFFER_SIZE` should have a `depends on`, will add it.

File: subsys/bluetooth/controller/hci/hci.c [hunk 2]
```diff
@@ -1156,6 +1162,32 @@ static void le_read_buffer_size(struct net_buf *buf, struct net_buf **evt)
 	rp->le_max_num = CONFIG_BT_CTLR_TX_BUFFERS;
 }
 
+#if defined(CONFIG_BT_CTLR_ADV_ISO) || defined(CONFIG_BT_CTLR_CONN_ISO)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/hci/hci.c:1165]: @Thalley Should this be as below? Does host built for only Synchronised Receiver use LE Read Buffer Size v2 command?

```suggestion
#if defined(CONFIG_BT_CTLR_ISO)
```

File: subsys/bluetooth/host/hci_core.c [hunk 3]
```diff
@@ -2423,27 +2423,27 @@ static void read_buffer_size_v2_complete(struct net_buf *buf)
...
 	if (!bt_dev.le.acl_mtu) {
 		return;
 	}
 
-	BT_DBG("ACL LE buffers: pkts %u mtu %u", rp->acl_max_pkt,
+	BT_DBG("ACL LE buffers: pkts %u mtu %u", rp->acl_max_num,
 		bt_dev.le.acl_mtu);
 
-	max_num = MIN(rp->acl_max_pkt, CONFIG_BT_CONN_TX_MAX);
+	max_num = MIN(rp->acl_max_num, CONFIG_BT_CONN_TX_MAX);
```
Comment (joerchan) [inline subsys/bluetooth/host/hci_core.c:2434]: Unrelated to this change.

@Thalley Why is it min with CONN_TX_MAX here? It is not like that for v1, so looks quite suspicious.

Comment (Thalley) [reply subsys/bluetooth/host/hci_core.c:2434]: @joerchan I guess in case of a non-Zephyr controller, we might expect `CONFIG_BT_CONN_TX_MAX` in the host, but but will need to handle it if the controller returns a smaller value, or even vice versa. 

If controller responds with 5 and CONFIG_BT_CONN_TX_MAX=10, then the maximum number of buffers we can use is 5, and vice versa if the controller responds with 10 and CONFIG_BT_CONN_TX_MAX=5, then we have only allocated 5 buffers in the host I suppose.

File: subsys/bluetooth/controller/hci/hci.c (hunk 2 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/hci/hci.c:1165]: ok

File: subsys/bluetooth/controller/hci/hci.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/hci/hci.c]: I have remove the `if defined(...` checks and added the dependencies in Kconfig. 

Not going to do anything for ACL buffers wrt `CONFIG_BT_CONN` there is a different PR that removes controller Kconfig related to ACL Tx count and buffer size. 

I will rebase this PR after #34771 is merged.

File: subsys/bluetooth/controller/hci/hci.c [hunk 4]
```diff
@@ -3248,6 +3280,12 @@ static int controller_cmd_handle(uint16_t  ocf, struct net_buf *cmd,
 		le_read_buffer_size(cmd, evt);
 		break;
 
+#if defined(CONFIG_BT_CTLR_ADV_ISO) || defined(CONFIG_BT_CTLR_CONN_ISO)
+	case BT_OCF(BT_HCI_OP_LE_READ_BUFFER_SIZE_V2):
+		le_read_buffer_size_v2(cmd, evt);
+		break;
+#endif /* CONFIG_BT_CTLR_ADV_ISO || CONFIG_BT_CTLR_CONN_ISO */
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/hci/hci.c:3277]: LE Read Buffer Size v2 is needed only when ISO feature is supported.

Cannot use `IS_ENABLED()` in switch-case, else will need to use `else` that returns unsupported command return code. Buy may be compiler is intelligent to optimize the else path?

[the remark this answers is not in the document]
Comment (Thalley) [reply subsys/bluetooth/controller/hci/hci.c:3277]: Ah, right, that's a good point. Didn't consider that :)

**Review by Thalley — APPROVED** [review]

**Comment by carlescufi** [comment]
@jhedberg can you please take another look?

**Review by jhedberg — APPROVED** [review]
